sdks/python/apache_beam/transforms/periodicsequence.py

```diff
@@ -37,6 +37,11 @@
 class ImpulseSeqGenRestrictionProvider(core.RestrictionProvider):
   def initial_restriction(self, element):
     start, end, interval = element
+    if isinstance(start, Timestamp):
```

Comment:
Should they be always Timestamps to avoid isinstance checks?

Reply:
I guess that would also work, I just didn't want to take the option of using other types, since some users may already be using them.

Reply:
Reviewing the code, our current tests use `time.time()` which return a float, I think we should not change it to only allow `Timestamps`. What do you think?

Reply:
Sounds good. Let's keep it as is.